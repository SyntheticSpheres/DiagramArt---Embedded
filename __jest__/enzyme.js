// https://github.com/airbnb/enzyme/issues/1265

const Enzyme = require('enzyme');
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });
