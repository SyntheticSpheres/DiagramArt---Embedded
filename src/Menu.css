.da-navbar {
    height: 50px;
    top: 0;
    left: 0;
    right: 0;
    background-color: #f5f5f2;
    padding: 0 10px;
}

.da-navbar ul {
    list-style: none;
    padding: 0;
    margin: 0;
    float: left;
}

.da-nav-right {
    float: right !important;
}

.da-nav-form {
    margin-top: 10px;
    margin-bottom: 10px;
}

.da-nav-form .da-form-input {
    padding: 6px 12px;
    outline: 0;
    font-size: 11px;
}
