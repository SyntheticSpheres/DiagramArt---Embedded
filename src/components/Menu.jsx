import React, {Component} from 'react';
import Settings from '../constants/Settings.js';
import DiagramListActions from '../actions/DiagramListActions.js';

/**
 *
 * @author Matej 'juffalow' Jellus <juffalow@juffalow.com>
 */
class Menu extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            query: ""
        };
    }

    handleQueryChange(event) {
        this.setState({query: event.target.value});
        DiagramListActions.loadLibrary(event.target.value, 1, 10);
    }

    render() {
        return (
            <nav className="da-navbar">
                <ul>
                    <li><a href="https://diagramart.com" target="_blank"><img src="https://cdn.shopify.com/s/files/1/0255/1589/t/7/assets/diagramart_logo_with_tm.png?14105803135550891160" alt="diagramart logo" height="50" /></a></li>
                </ul>
                <form className="da-nav-form da-nav-right">
                    <input type="text" className="da-form-input" value={this.state.query} onChange={this.handleQueryChange.bind(this)} placeholder={Settings.language.search} />
                </form>
            </nav>
        );
    }
}

export default Menu;
