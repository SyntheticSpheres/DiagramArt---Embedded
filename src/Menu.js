import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Menu.css';

class Menu extends Component {
  constructor(props) {
    super(props);

    this.state = {
      query: '',
    };

    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(event) {
    this.setState({query: event.target.value}, () => { this.props.onQueryChange(this.state.query); });
  }

  render() {
    return (
      <nav className="da-navbar">
        <ul>
          <li><a href="https://diagramart.com" target="_blank" rel="noopener noreferrer"><img src="https://cdn.shopify.com/s/files/1/0255/1589/t/7/assets/diagramart_logo_with_tm.png?14105803135550891160" alt="diagramart logo" height="50" /></a></li>
        </ul>
        <form className="da-nav-form da-nav-right">
          <input type="text" className="da-form-input" value={this.state.query} onChange={this.handleChange} />
        </form>
      </nav>
    );
  }
}

Menu.propTypes = {
  onQueryChange: PropTypes.func.isRequired,
}

export default Menu;
