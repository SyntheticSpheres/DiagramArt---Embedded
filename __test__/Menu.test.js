import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import sinon from 'sinon';
import Menu from '../src/Menu';

describe('Menu component', () => {
  let sandbox;
  let callback;

  beforeEach(function() {
    sandbox = sinon.sandbox.create();
    callback = sandbox.stub();
  });

  afterEach(function() {
    sandbox.restore();
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Menu onQueryChange={callback} />, div);
  });

  it('renders correctly', () => {
    const tree = renderer.create(
      <Menu onQueryChange={callback} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('simlate change', () => {
    const tree = shallow(
      <Menu onQueryChange={callback} />
    );
    tree.find('input').simulate('change', { target: { value: 'q' } });
    sinon.assert.calledOnce(callback);
    sinon.assert.calledWith(callback, 'q');
  });
});
