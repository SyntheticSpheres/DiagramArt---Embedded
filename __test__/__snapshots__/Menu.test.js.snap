// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component renders correctly 1`] = `
<nav
  className="da-navbar"
>
  <ul>
    <li>
      <a
        href="https://diagramart.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="diagramart logo"
          height="50"
          src="https://cdn.shopify.com/s/files/1/0255/1589/t/7/assets/diagramart_logo_with_tm.png?14105803135550891160"
        />
      </a>
    </li>
  </ul>
  <form
    className="da-nav-form da-nav-right"
  >
    <input
      className="da-form-input"
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
</nav>
`;
